react/features/base/connection/actions.native.js

```diff
@@ -15,6 +15,53 @@ import {
     SET_LOCATION_URL
 } from './actionTypes';
 
+/**
+ * The error structure passed to the {@link connectionFailed} action.
+ *
+ * Note there was an intention to make the error resemble an Error instance (to
+ * the extent that jitsi-meet needs it).
+ */
+export type ConnectionFailedError = {
+
+    /**
+     * The invalid credentials that were used to authenticate and the
+     * authentication failed.
+     */
+    credentials?: {
+
+        /**
+         * The XMPP user's ID.
+         */
+        jid: string,
+
+        /**
+         * The XMPP user's password.
+         */
+        password: string
+    },
+
+    /**
+     * The details about the connection failed event.
+     */
+    details?: string,
+
+    /**
+     * Error message.
+     */
+    message?: string,
+
+    /**
+     * One of {@link JitsiConnectionError} constants (defined in
+     * lib-jitsi-meet).
+     */
+    name: string,
```

Comment:
this looks quite weird to me while reading it. why did you choose to call it "name" instead of "error" ?

Reply:
I didn't, it was like that. If you check on the docs above it mentions that the structure tries to be similar to JS Error class:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name

Reply:
Ah, ok.